\chapter{Abstract}                                 \label{ch:abstract}

%\ldots

\instructionsabstract

% Transcriptome - microarray - compendium

With the unprecedented ability to systematically probe gene expression at genome scale, microarray has become an indispensable technology adopted by every laboratory across the world, generating a colossal amount of data for a variety of species. Although comprehensive at gene dimension, any microarray based study alone provides a limited scope to study specific aspect(s) of a complex organism through analyzing its transcriptome. Alternatively, the aggregate of existing expression data provides an opportunity to investigate gene expression of a particular species at a global level, and alternatively, view a specific study from the perspective of existing knowledge. The goal of this research focuses on developing novel methodology and systems to explore this opportunity.

We first developed a methodology to create an organism-specific cross-platform compendium from publicly available gene expression data. Special attention has been paid on two aspects, resolving data representation heterogeneity to facilitate automated data retrieval across data generated on different microarrays, and improving data consistency and compatibility through systematic renormalization of data and subsequently, log-ratio calculation. Compared with the existing approach that constructs compendium from the data generated on single-platform, this novel methodology is superior with its broader applicability and the extensive scope of the resulting compendium.

Utilizing this novel methodology, we constructed three such comprehensive expression compendia for the bacterial model organisms (\textit{Escherichia coli}, \textit{Bacillus subtilis}, and \textit{Salmonella enterica} serovar Typhimurium). Moreover, efforts have been taken to create a web access portal providing public access to these three compendia. Especially, a suite of intuitive tools suitable for exploring, analyzing, and visualizing expression data of such cross-platform compendium are incorporated into this web portal to promote the utility of the compendium. Employing these tools to explore \textit{E. coli} compendium, we successfully identified novel targets for transcription factor FUR (Ferric Uptake Regulation) based on their expression similarity to that of the known FUR targets across a range of conditions of different experiments and platforms origins.

One of the most important applications of compendium to study a organism's response to the genetic and environmental changes by identifying condition dependent functional modules, and studying the underlying regulatory mechanism responsible for the observed expression variations. Many methods exist for this purpose. Each makes different assumptions to handle the under deterministic nature of this complex problem, and consequently generates complementary results. Here, we demonstrated such complementarity between two methods, DISTILLER and COLOMBOS, in a case study, in which co-expression modules containing gene \textit{sodA} are extracted from \textit{E. coli} compendium using each method, and compared against each other. Through this example, we stress the importance of choosing the right method based on the research purpose.

At last, we extended the methodology to handle the extended complexity of monocot \textit{Zea mays}, specifically addressing two issues, platform probe annotation inconsistency and precise biological sample annotation that manifests the abundant genetic repository (breeding lines) of maize species, and its complex life style (development stage) and structure (tissue). We also upgraded web access portal accordingly with new functions to facilitate the utility of rich annotations available.




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Keep the following \cleardoublepage at the end of this file, 
% otherwise \includeonly includes empty pages.
\cleardoublepage

% vim: tw=70 nocindent expandtab foldmethod=marker foldmarker={{{}{,}{}}}
