\chapter{MAGIC: access portal to a cross-platform gene expression compendium for maize}\label{ch:magic}
\chaptermark{Magic}

\ldots

\instructionsintroduction

% NOTE: no abstract for each section, otherwise mess up page numbering!!
%\textbf{Abstract}
%To facilitate the exploration of publicly available Zea mays
%expression data, we constructed a maize expression compendium,
%making use of an integration methodology and a consistent probe to
%gene mapping based on the 5b.60 sequence release of Zea mays. The
%compendium is made available through a web portal MAGIC that
%hosts a variety of analysis tools to easily browse and analyse the
%data. Our compendium is different from previous initiatives in combining 
%expression values across different experiments by providing a consistent 
%gene annotation across different platforms.



\section{Introduction}
Owing to its importance as sustainable food and feedstock,
maize genomics is of high academic and industrial relevance.
As a result, microarrays have been widely applied to interrogate
the maize transcriptome, with currently over one hundred maize gene expression 
experiments being publicly available in online repositories
such as GEO \cite{Barrett2011} and ArrayExpress \cite{Parkinson2009}. 
However, cross-platform differences, the lack of
consistent platform and measurement descriptions, and inconsistent 
gene annotations complicated the straightforward
use of these data.

To integrate the data from different array platforms in a readily 
usable single compendium, we resolved gene annotation
inconsistencies by reannotating probes of previously published
Zea mays arrays using the published maize genome sequence
(Schnable et al., 2009) and made measurements comparable
across different platforms/experiments using an adapted version
of the data integration method described by Engelen et al. (2011).
This resulted in a cross-platform expression compendium containing 
1749 microarrays covering 24690 genes. Additional
gene information was integrated from various external sources.
Experimental annotations were manually curated. A web access
portal MAGIC with a specialized set of exploration and analysis
functionalities provides public access to this compendium.


\section{Materials and Methods}

\subsection{Compendium creation}

The compendium itself corresponds to a matrix of which the rows correspond 
to genes and columns to sample contrasts. A sample contrast is
defined as the comparison of the gene’s expression between two different
biological samples, one acting as a test and the other as a reference.Each
value in the matrix then represents the expression change of a gene presented 
as the log-ratio of its expression in the test versus the reference
samples.

\subsubsection{Probe reannotation} Probes were reannotated using the latest
release of the maize genome. Original probe sequences, if available, were
obtained from the respective platform annotation files or otherwise from
GenBank based on the corresponding GI numbers. They were used as
query against the curated ‘Filtered Gene Set’ (FGS) of the 5b.60 release
of B73 maize genome using MegaBLAST version 2.2.17 (Zhang et al.,
2000). Both the gene and the transcript models of FGS are searched to
increase the eventual hit rate. Different blast parameters were chosen for
oligo and cDNA probes, respectively, as they considerably differ in length
and specificity. For probes that mapped to multiple genes, we identified
the most specific hit by comparing the hit quality across different targets.
Only probes for which a sufficiently unique probe-to-gene mapping (details 
in Section \ref{ch:apd-magic}) could be identified were retained for
compendium construction.

\subsubsection{Expression data homogenization}

Microarray experiments were retrieved from GEO and ArrayExpress. Each 
experiment can be composed of multiple arrays. 
A distinction is made between one- and two-channel arrays, which differ from 
each other in the number of samples tested per array. 
Irrespective of the used platform, raw expression intensities were extracted 
for each channel (sample) separately. 
To make measurements between one- and two-channel platforms comparable, all
expression measurements were converted to log ratios (as contrasts) that
compare the expression between a test and a reference sample, respectively, 
both from the same experiment. 
For each contrast, proper test and reference samples were assigned based on the 
corresponding experiment annotation. 
Raw expression data were subsequently normalized using dedicated procedures 
(\cite{Engelen2011}) and mapped to the corresponding genes based on the probe 
annotation (see Section 2.1.1). 
Special strategies have been developed to handle data generated using 
Affymetrix platform and data generated on multiple-chip 
platforms\footnote{Multiple-chip platform are multiple microarray chips that 
are designed together with their probes targeting complementary gene sets, and 
that are used in combination to interrogate the same biological sample in order 
to measure the expression levels of more genes than would be possible with 
only one chip.}. 
The details are explained in Section \ref{apd:magic-datanorm}.


\subsection{Compendium annotation}

To improve biological interpretation of the compendium, we integrated
gene (row) and contrast (column) annotation from publicly available 
resources and curated all available information.
To facilitate gene selection, we included, next to gene ids from 5b.60
genome release, gene names fromMaizeGDB (Lawrence et al., 2004)and
Xref assignments from www.maizesequence.org to provide mapping from
EntrezGene and UniProt ids. As for functional annotations, metabolic
pathway information (version 2.0) from Gramene (Youens-Clark et al.,
2010) and Gene Ontology annotations from www.maizesequence.org
were provided.

To compensate for the often cryptic and incomplete condition annotations 
available in public expression repositories, we provided curated
annotations incorporating the information from both online repositories
and the corresponding publications. Note that in our compendium, expression 
values are represented as log-ratios of a contrast between two
samples. Experimental annotation is provided both at the level of the
individual sample and that of the contrast. Annotation at the sample
level includes tissue, development stage and genotype specifications
(breeding line). The first two are described using Plant Ontology
(Avraham et al., 2008)-derived ontology terms, whereas genotype specifications 
are based on the names of cultivars or wild-type. At the contrast
level, we associated perturbation annotations specified as a set of relevant
properties and corresponding values of change that reflect the stimuli that
trigger expression alterations in the test versus the reference.





\subsection{Compendium exploration}

To facilitate the exploration of the compendium, we constructed a web
access portal MAGIC providing a set of analysis functionalities. At first,
MAGIC allows users to specify their own subcompendium of interest,
which only contains contrasts sharing the same characteristics. Users can
choose between various predefined subcompendia. Subcompendia focusing 
on environmental perturbations and on comparisons between lines,
between tissues and between development stages are available.
Alternatively, users can generate customized subcompendia based on
the sample and contrast annotations.

Once a (sub)compendium is selected, the system provides tools to explore 
and visualize the expression data in a module-centralized manner
where a module is defined as a subset of the (sub)compendium containing
the expression values of a set of genes under a set of contrasts. A module
can be created starting from a query set of genes or contrasts to which
contrasts or genes are added, respectively, based on their properties (such
as the coexpression level, or the expression consistency). An existing
module can easily be altered, merged with other modules or split into
several modules. Each module can be visualized as an interactive heatmap
accompanied with corresponding annotation information.


\subsubsection{Missing value handling}

Most of the microarray platforms that MAGIC relies on were developed before the 
genome release of maize. 
None of them cover the full FGS gene set, and overlap in measured genes can be 
low for some platform combinations (Supplementary Tables 
\ref{tab:maize-platform-overview} and 
\ref{tab:maize-platform-overlap}).
Consequently, each contrast contains some missing values, as not all genes are 
measured in the corresponding platforms.

Several strategies are employed to tackle this issue.  While creating the 
compendium, FGS genes without any measurement are removed.  
We further extended the web port with extra functionalities that help users to 
evaluate and control the number of missing values in their analysis results.
%
In every module creation and modification functions that select(or remove) 
genes or contrasts, the missing value information is provided for each 
candidate, and can be utilized to filter out those having too few measurements.
%
However, under certain situation, it is not possible to directly control the 
candidates based on missing values, for example, manually adding genes into a 
module.
%
Alternatively, the percentage of the missing value is calculated for each 
module.
User can consult this information and take further action if necessary.
%
By integrating these functionalities into the web portal, the amount of missing 
value allowed in the user data can be easily contained.
%
A detailed explanation for those functionalities are provided in an 
online help section dedicated to this issue. 



%\textbf{reminder}
%
%In practical use, we noticed that the default contrast selection criterion 
%"inverse coefficient of variance (iCV)" statistics previously used tends to 
%select contrasts where most candidates does not show any differentiated 
%expressions instead of those where a fraction of the candidates show strong 
%changes, as its value deteriorates rapidly with a small portion of genes with 
%deviated expression pattern from the majority whereas retains high for those 
%contrasts where most candidates has no differential expressions consistently.
%
%The matter is exacerbated further by two factors. First, there is a large 
%amount of missing values. The more values are missing, the more consistent the 
%expressions of the candidates can be as less values are used to calculate the 
%standard deviations used to calculate iCV. Second, most maize pathways or GO 
%annotations are connected to large number of genes potentially with 
%diversified 
%functionalities and expression patterns... 





\section{Results and Discussion}

The compendium available through MAGIC currently contains
24690 genes and 1310 sample contrasts. It covers 62% of the
genes in FGS of the 5b.60 release of B73 maize genome; the
remaining genes are not represented on any of the 27 platforms
included. The contrasts consist of 69 experiments obtained from
GEO and ArrayExpress, amounting to 1749 microarrays. Details
on the composition of the compendium in terms of the number
of genes and experiments covered per platform can be found in
Tables \ref{tab:maize-platform-overview} and 
\ref{tab:maize-exp-overview}. 
On average, a gene is measured in 9 of the 26 platforms and has been measured 
in 592 of 1310 contrasts (Figures \ref{fig:maize-accu-platform-count} and 
\ref{fig:maize-accu-contrast-count}). 
The compendium can be downloaded through its web portal MAGIC. 
An elaborate online help, together with two tutorial case studies,
illustrates how the various functionalities of MAGIC can be
used to infer new biology. 
The case studies clearly illustrate how information from different platforms 
contributes to the results obtained by MAGIC, and how to cope with missing 
values that could become abundant if information from certain sets of
platforms is combined.

\todo{[magic]A table to compare diff. method}
In contrast to MAGIC, comparable initiatives treat data from
different platforms or experiments separately. Genevestigator
(Hruz et al., 2008) and CORNET (De Bodt et al., 2012) construct 
separate compendia for the Affymetrix Maize Genome
Array (GPL4032) (containing 558 and 340 arrays for
Genevestigator and CORNET) and the Nimblegen Maize
385 k Array (GPL12620) (containing 180 arrays in both systems).
PLEXdb (Dash et al., 2011), on the other hand, provides access
to the data from 44 Affymetrix and Nimblegen experiments. In
this system, data derived from each experiment are treated separately 
instead of being merged in a larger compendium. Of all
these systems, only CORNET provides a restricted meta-analysis
tool that allows combining information across the different compendia.
Compared with these related initiatives, our approach is
unique in directly combining data from different platforms in a
single compendium, obviating the need for an additional meta-analysis 
step (Fierro et al., 2008) and enabling the construction
of a much larger compendium and the direct data analysis across
different platforms and experiments.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Keep the following \cleardoublepage at the end of this file, 
% otherwise \includeonly includes empty pages.
\cleardoublepage


% vim: tw=70 nocindent expandtab foldmethod=marker foldmarker={{{}{,}{}}}
