\chapter{Directed module detection in a large-scale expression compendium}\label{ch:distiller}
\chaptermark{Gene coexpression module discovery}

% \ldots

% \instructionsintroduction

% specially indented text in bold 
\newcommand{\distsubparagraph}[1]{\hspace{-2mm}\textsf{\textbf{#1}}}

\todo{Check all the links and update them}

\todo{Check and add all abbreviations and symbols}
\glossary{name=DISTILLER, description=Data Integration System to Identify Links 
in Expression Regulation}
\glossary{name=ViTraM, description=VIsualization of TRAnscriptional Modules}

\glossary{name=EcoCyc, description=Encyclopedia of Escherichia coli K-12 Genes 
and Metabolism}
\glossary{name=BioCyc, description=Pathway/Genome Databases and Pathway Tools 
Software}
\glossary{name=RegulonDB, description=The reference database of the regulatory 
network of \textit{Escherichia coli} K-12}
\glossary{name=UniPort, description=Universal Protein Resource}

\glossary{name=PSSM, description=Position Specific Scoring Matrices}

\glossary{name=ChIP, description=Chromatin Immunoprecipitation}


\nomenclature{Compendium}{Essentially an expression (log-ratio) value matrix, 
whose rows correspond to the known genes and columns correspond to sample 
contrasts}
\nomenclature{Sub-compendium}{A subset of compendium containing contrasts 
sharing certain properties of interests, e.g., contrasts in which the gene 
expression profiles of different tissues are compared}
\nomenclature{Sample contrast}{Where the expression values of genes from two 
biological samples, one set at the test and the other reference, are compared 
and the expression variations between samples are represented as log ratio}
\nomenclature{Coexpression}{Genes' coherent expression responses to 
certain alterations in the organism's intra- or extracellular environment}
\nomenclature{Coexpression Module}{A combined set of genes and condition 
contrasts, where the coexpression pattern appears}
\nomenclature{Homogenization}{Our methodology to convert raw 
intensity expression values from individual biological sample into expression 
log-ratios between two different biological samples paired as a contrast so 
that they are comparable across different microarray platforms and experiments}


%\textbf{Abstract}
%Public online microarray databases contain tremendous amounts of expression 
%data. Mining these data sources can provide a wealth of information on the 
%underlying transcriptional networks. In this chapter, we will illustrate how 
%the web services COLOMBOS and DISTILLER can be used to identify condition 
%dependent coexpression modules by exploring compendia of public expression 
%data. COLOMBOS is designed for user-specified query-driven analysis, whereas 
%DISTILLER generates a global regulatory network overview. The user is guided 
%through both web services by means of a case study in which condition 
%dependent coexpression modules comprising a gene of interest (i.e. `directed') 
%are identified.

\section{Introduction}

Omics based approaches are increasingly being used to uncover underlying 
mechanisms of bacterial behaviour \cite{Fierro2008}. The obligate deposit of 
high throughput experiments in public databases upon publication has 
tremendously increased the amount of publicly available experiments. Mining 
these data sources helps in gaining a global condition dependent view on 
bacterial gene regulation \cite{Lemmens2009, Fadda2009}, in expanding the 
current knowledge on transcriptional interactions with novel reliable 
predictions \cite{Faith2007, Ernst2008}, and in comparing transcriptional 
networks across species \cite{Fierro2008, Babu2009}. It also offers molecular 
biologists the possibility to see their own dedicated analysis in light of what 
is already available. Inferring transcriptional networks from these public data 
usually requires complex normalization procedures \cite{Faith2008} and 
computationally intensive algorithms. To enhance the usability of these tools, 
they are often wrapped in a web service.

In this book chapter, we will illustrate how compendia of public expression 
data can be used to identify condition dependent coexpression modules, in which 
a particular gene of interest is involved (`directed' module detection), by 
means of two web services: COLOMBOS \cite{Engelen2011} and DISTILLER 
\cite{Lemmens2009} (the latter in combination with the visualization tool 
ViTraM \cite{Sun2009}). Figure \ref{fig:workflow-distiller-colombos}
illustrates the difference between both approaches. COLOMBOS only relies on 
expression data to retrieve condition dependent modules, implying that there is 
a functional relationship between the module genes, but not necessarily that 
they are regulated by the same (set of) transcription factor(s). DISTILLER, in 
contrast, incorporates additional motif data and the constraint that module 
genes should share the same regulatory program, implying that the module's 
condition dependent coexpression can be directly linked to transcriptional 
coregulation. 

To clearly demonstrate the functionalities of these web services, 
the gene \textit{sodA} is used as a case study. This gene encodes for a protein 
with superoxide dismutase activity, which reduces harmful free radicals of 
oxygen formed during normal metabolic cell processes \cite{Fawcett1995, 
Tardat1993, Jair1995}. 
It is known to be regulated by several regulators, such as, 
Fur, SoxS, and MarA. As such its expression is coupled to different biological 
conditions: multiple antibiotic resistances (MarA), superoxide resistance 
(SoxS) and the intracellular iron pool (Fur). By applying different analysis 
approaches (COLOMBOS and DISTILLER), we will show how to identify \textit{sodA} 
containing modules, i.e. genes that behave similarly to \textit{sodA} in a 
condition dependent way from large-scale expression compendia.

\begin{figure}[tb]
	\centering
  	\includegraphics[width=1\textwidth]{Fig1-Workflow-figure.png}
	\caption[Workflow of directed module detection in expression 
	compendium]{
	\textbf{Workflow of different methods for the directed module detection in 
	expression compendium.} 
	Coexpression modules are detected by COLOMBOS starting from a user provided 
	seed gene (e.g., \textit{sodA}). 
	DISTILLER searches for coexpression modules in a global fashion 
	(information on seed genes is not a prerequisite) and is constrained by 
	regulator-to-gene assignments. 
	Coexpression patterns retrieved by COLOMBOS point toward a functional 
	relationship among the module genes but do not necessarily imply 
	coregulation. To gain insights on the regulatory program of those genes, de 
	novo motif detection tools could be applied to analyze their promoter 
	regions. Inherent to the regulator-to-gene constraints implemented in 
	DISTILLER, there is a direct link between coexpression and coregulation for 
	the genes in the modules that it retrieves.}
	\label{fig:workflow-distiller-colombos}
\end{figure}

\section{Material}


\subsection{Cross platform expression compendium}\label{sec:dist-comp} 

The cross platform expression compendium consists of a large amount of publicly 
available microarrays derived from different platforms (e.g. cDNA microarrays, 
Affymetrix). 
Data were normalized to remove cross platform differences and 
experimental conditions were annotated in a formal manner (see Section 
\ref{sec:colombos-comp}).
The expression data are presented as expression log ratios of measurements in a 
test condition versus a reference condition.
A \textit{condition contrast} is the set of all 
genes' expression log ratios, each of which represents the log ratio of a 
gene's expression values obtained in one particular experiment for which a test 
is compared to a reference conditions. 
The condition annotations of each contrast thus indicate which aspect(s) 
(experimental conditions) has/have been changed between the test and reference 
samples.  
Please refer Section \ref{sec:colombos-comp} for more details about 
the compendium and its condition annotations.

The \textit{Escherichia coli} compendium used in here  
contains 1429 condition contrasts obtained from 1747 microarrays of 84 
experiments across 35 platforms, covering the expression profiles of 4295 genes 
under various conditions annotated by 242 different condition properties 
grouped under 56 condition ontology terms.
Furthermore, several sources of gene information from main curated databases 
are integrated into the compendium as well, such as, the pathway and operon 
information from EcoCyc \cite{Keseler2009}, the Gene Ontology annotation from 
UniProt GOA \cite{Camon2004}, and the regulon information from RegulonDB 
\cite{Gama-Castro2008}. (see Table \ref{tab:colTB-overview})

%The condition annotation is specified using a set of formal condition 
%properties (representing for instance mutations, compounds in the growth 
%medium, treatments, and general growth conditions) and associated absolute 
%values for these properties (for instance, concentration of a certain compound 
%in the growth medium). As such, we can specify the condition annotation of 
%each 
%condition contrast rigidly as a vector representing the differences between 
%these property values in test and reference condition for each of the relevant 
%condition properties (for instance, the concentration difference between test 
%and reference condition). This enables the mathematical comparison of contrast 
%annotations. For instance, a contrast condition annotation specified as 
%`FeSO$_4$: 50$\upmu$M' describes that for this condition contrast, the 
%concentration of FeSO$_4$ has changed by a quantity of `50$\upmu$M', in the 
%medium of the test versus the reference sample. Due to the large number of 
%different properties and their distinct nature, it may not be easy for a user 
%to find those condition properties he or she is interested in. To facilitate 
%this, we have provided a condition ontology similar to Gene Ontology 
%\cite{Ashburner2000}. This ontology maps the condition properties used to 
%annotate the condition contrasts to the biological processes or 
%functionalities 
%they most likely affect. Each property can be 
%mapped to one or more ontology terms and thus ensures an intuitive, higher 
%level view of the condition contrasts. As an example, the condition ontology 
%term `amino acid metabolic process' includes condition properties that are 
%linked to amino acid metabolism, such as the addition of arginine to the 
%medium, \textit{trpR} mutation (tryptophan transcriptional repressor), etc.  
%The condition ontology superimposed on the condition annotation thus 
%represents 
%a broader, more intuitive view of the altered stimuli that are responsible for 
%the observed expression changes (i.e. log ratios) of the condition contrasts. 


\subsection{Coexpression modules}

Genes that have the same regulatory program (here defined as having the same 
transcription factor binding motifs in their promoter regions) behave 
similarly: their expression responds in a similar manner to certain alterations 
in the organism's intra- or extracellular environment. This is called 
\textit{coexpression}. The evidence of coexpression points to a functional 
relationship between coexpressed genes, and might sometimes be an indication of 
shared regulatory programs. 
Such a functional relatedness is only enforced when 
coexpression occurs across multiple different conditions. Since each value in 
our data set does not represent an absolute measure of expression, but rather a 
relative one associated to a condition contrast (test versus reference), 
coexpression in this case translates to genes showing log ratios that are 
significantly different from zero for at least one condition contrast, and for 
each of the involved contrasts show coherent changes in the same direction 
(either up or down). Both methods that we apply in this chapter, COLOMBOS and 
DISTILLER, employ different computations to score this coexpression, but 
essentially look for the same phenomenon.

Notice that genes are only coexpressed under certain condition contrasts. This 
combined set of genes and condition contrasts, where the coexpression pattern 
appears, is what we refer to here as a \textit{module}. Extra constraints can 
be used to tune the module concept for specific purposes. For example, the 
requirement could be added that genes in the module should have some (known) 
motifs in common.


\subsection{COLOMBOS}
COLOMBOS, an acronym for \textit{COLections Of Micrcroarrays for Bacterial 
OrganismS}, is a web service \cite{COLOMBOS} (Fig. \ref{fig:colombos}) for 
interactively exploring, querying, analyzing and visualizing data from cross 
platform expression compendia through an intuitive interface. 
Currently it 
provides the possibility to query expression compendia based on the annotation 
of their experimental condition contrasts and/or genes. COLOMBOS can be used 
for instance to search for genes being coexpressed with one or more genes of 
interest under a pre-specified set of condition contrasts, or to search for the 
condition contrasts in an expression compendium under which a pre-specified 
gene set is coexpressed. As such, it can identify coexpression modules based on 
user input. In COLOMBOS, modules can be visualized as interactive heatmaps, 
showing the annotation of genes and condition contrasts as obtained from public 
databases and corresponding literature. More details of COLOMBOS are discussed 
in Chapter \ref{ch:colombos}.

\begin{figure}[tb]
	\centering
  	\includegraphics[width=1\textwidth]{Fig2-COLOMBOS.png}
	\caption[COLOMBOS web service interface]{\textbf{COLOMBOS web 
	service interface.} 
	The interface is composed of three parts: the website navigation header at 
	the top, the workspace (showing a module tree) at the left-hand side, and 
	the operating space (here showing the `Data analysis overview' panel) 
	filling the center.}
	\label{fig:colombos}
\end{figure}

Currently COLOMBOS (v1.0) runs on three browsers: Firefox, Google Chrome, and 
Opera. Be sure that you are using the latest version of the browser to have the 
best compatibilities. Please check the website for updated information. 
% Please refer to \textbf{Note 1} for version and supported browsers of the 
%website.

\subsubsection{COLOMBOS expression data format}\label{sec:dist-format-col}
As the expression compendium is an integrate part of the COLOMBOS web service, 
no input files are required to use it.  
It does provide a tab delimited ASCII file format for expression data export, 
containing information on the user created module consisting of a particular 
set of genes and the corresponding set of condition contrasts under which these 
genes are coexpressed. 
There are two sections in this file. The first one describes the 
condition information of the module. It specifies for each condition contrast 
the name and description of its test and reference sample, the corresponding 
experiment identifier, the online database from which the experiment was 
retrieved, the microarray platform used to measure the sample hybridizations, 
and finally, the annotated contrast condition changes between the test and the 
reference sample. The second section contains three parts: the corresponding 
condition contrast identifiers in the first row; the gene information (gene 
locus tag, gene name, and internal gene identifier) in the first three columns; 
and the log ratio expression data from the expression matrix. Note that 
starting from the 4$^{th}$ column, each column in the file corresponds to one 
condition contrast. Each of the rows represents one gene and its expression 
values for different condition contrasts.


\subsection{DISTILLER}\label{sec:dist-distiller}
DISTILLER stands for \textit{Data Integration System To Identify Links in 
Expression Regulation}. It is a data integration framework that searches for 
condition dependent transcriptional modules by combining expression data with 
information on the interaction between a regulator and its corresponding target 
genes, for instance based on motif screening. A module is here defined as a set 
of genes that are both coexpressed across a sufficiently large number of 
columns in the expression matrix and that share motif instances for the same 
regulator(s). Note that the expression data used by DISTILLER could be either 
absolute expression values or log ratios. 

DISTILLER searches the modules in a global fashion over the entire dataset 
through three steps: identification candidate modules, module filtering, and 
module extension. 
Built on advanced itemset mining approaches, it first 
exhaustively enumerates all possible valid \textit{closed} modules (see next 
Section) as candidates. The candidates are partially redundant as 
they might share the same genes or condition contrasts. 
Therefore, in the second step (filtering), a module prioritization is provided 
in which an interest score is assigned to each module. The score takes into 
account the probability that a module of the same size can be found by chance 
in the input dataset, but at the same time, penalizes the overlaps with those 
modules that have already been reported. The top ranking modules are 
statistically significant yet distinctive, and thus selected as the output of 
this step. 
These initial modules obtained by DISTILLER are usually called `\textit{seeds}' 
as they are obtained under stringent thresholds. 
This has the advantage that all identified interactions between a transcription 
factor and its target genes are highly reliable (high precision), but that true 
interactions might be missed (lower sensitivity).  
Hence, in the last step (module extension), the obtained 
modules can be further extended with additional genes by relaxing the used 
thresholds on the required minimal coexpression level and motif score.  Note 
that it is not required to execute all three steps. 
The result of each step can be analyzed and visualized by the user 
independently.

The DISTILLER web service \cite{DISTILLER} (Fig. \ref{fig:distiller}) is 
designed to leverage the difficulty of applying the algorithm by providing an 
easy-of-use and consistent interface that links all three steps together. 
The interface is separated into two sections vertically. The information panel 
at the top provides a brief explanation of the site's functionality. The 
project panel at the bottom is where the user interacts with the site to use 
our DISTILLER service. 
The panel is further divided into three sections horizontally.  At left, the 
`Project Information' panel provides information of the current project and the 
links to access the input, output, and log files.  
The `Operation' panel at the center is where user can run DISTILLER algorithm.  
It has three tabs, each of which provides access to one step of the
algorithm, namely, module detection, module filtering, module extension.  Based 
on the step selected in the `Operation' panel, the explanation for the 
corresponding parameter settings and the input data file format (if required) 
is provided in the `Online help' panel at right for your reference.
The website is guaranteed to work properly on Firefox and Google Chrome.
Efforts are undertaken to make it compatible with other browsers as well. 
Please check the website for updated information. % For compatibility issues, 
%please refer to \textbf{Note 1}.

\begin{figure}[tb]
	\centering
  	\includegraphics[width=1\textwidth]{Fig3-DISTILLER.png}
	\caption[DISTILLER web service interface]{\textbf{DISTILLER web service 
	interface (at module detection step).}}
	\label{fig:distiller}
\end{figure}

For this case study, the data consist of the same compendium as the one 
accessed through the COLOMBOS website. Hence, the columns of the expression 
matrix, across which DISTILLER defines coexpression, correspond to the 
condition contrasts as defined in the compendium (see Section 
\ref{sec:dist-comp}). 


\subsubsection{Closed module}
A module is considered \textit{closed} if it cannot be further extended by any 
other gene without reducing the number of motifs shared by all of its genes. To 
search such a candidate, the algorithm \cite{Lemmens2009} starts with the 
smallest module that contains only one gene, and gradually extends them by 
merging with other modules. A candidate is found when any extension violates 
the conditions implied by `closed'. 
Only closed modules that contain the minimal number of 
motifs and condition contrasts are considered valid (meet the required minimal 
Supports on the motifs and condition contrasts)

\subsubsection{DISTILLER input file format}\label{sec:distiller-infile-format}
DISTILLER requires two input files: an expression data file and a motif data 
file. In the case study described here, the expression data file contains the 
log ratio expression values for each gene under each condition contrast. Two 
possible input formats are allowed for the expression data file. 
The first one is a matrix based flat file format, in which each row represents 
the data of one gene; each column represents a condition contrast in the 
expression compendium. 
The only exception is the first row, where the identifiers of each condition 
contrast are specified, with the first column either left empty or specified 
as `gene name'.  
The gene names are specified in the first column of each row.  
The second format that is permitted is the format of the expression data file 
that can be obtained for each user created module from COLOMBOS (see Section 
\ref{sec:dist-format-col}).
Due to the possible large size of the expression data set, the system further 
supports the expression data file in two compression types, namely, 
Zip\footnote{In case of zip file, the 
system allows only one file in the compressed file that is the input expression 
data file. An error is generated when this condition is violated.} and Gzip.


The motif data file contains the results of a motif screening analysis 
\cite{Hertzberg2005}. 
Each motif screening score indicates the probability that a motif instance is 
present in the promoter region of a gene, with the maximum score 1 and the 
minimum score 0. These data are also imported in a matrix format file, where 
the rows in the file represent the genes and the columns represent the motifs 
for which the screening was performed. Similarly to the expression data, the 
first row in the file contains the motif information with one motif name in 
each column, and from the second row on, each row represents one gene's motif 
scores with the corresponding gene name specified in the first column.

In addition to the two files mentioned above, users can provide an operon 
information file that will be used during the module extension step of 
DISTILLER. It contains two items in each row, the first one is the gene that 
belongs to an operon, and the second one is the first gene of that operon.

In DISTILLER, the expression and motif matrix are coupled in the gene 
direction, so the user should make sure that the same gene identifiers are used 
in both files (as well as in the operon information file, if included), 
although the order of the genes does not have to be the same.

\subsubsection{DISTILLER output file format}\label{sec:distiller-outfile-format}
For all three steps of DISTILLER, the outputs are 
of the same format: `.m' file.  The file contains the information of the 
identified modules, with each module described in a section, separated by an 
empty line.  Each section contains four data fields, each representing a part 
of the module information.  Given a section specifying the information for a 
module $M$, the field `Significances' holds the $p$-values that were assigned 
to $M$ by the itemset mining algorithm \cite{Zaki} used by DISTILLER.  The 
field `items' contains the information of the genes that belong to $M$, each of 
which is represented by its row number\footnote{When counting the row number 
of a given gene, there are certain caveats. First, the title line should not be 
counted. Second, the count should start from 0. Hence the first row is of row 
number 0 instead of 1.} in the expression data input file. The field 
`boxtidset1' specifies the contrasts of $M$, in which the genes in $M$ are 
coexpressed, each of which is represented by its column number\footnote{To 
count the column number referred in the `.m' file, the first column containing 
gene information should not be counted, and it starts at 0.}
% \footref{fn:dist-count} 
in the expression data input file. Finally, the field `tidset1' 
contains the motifs of $M$, which are shared by the genes in M, specified by 
their column numbers in the motif data input file. Each module is represented 
by a number indicated between the `\{\}' brackets after the name of each data 
field.

Besides the main output file, two supplementary files required to use the 
visualization software ViTraM are also provided. They are bundled in a 
compressed file that can be downloaded from the DISTILLER website. The file 
whose name starts with `expdata\_' contains the DISTILLER preprocessed input 
expression data, and the file whose name starts with `binary\_' contains the 
DISTILLER preprocessed input motif data. The file formats of these expression 
data and motif data files are the same as the corresponding input files 
described in previous Section.

\paragraph{Output notification email}\label{sec:distiller-email}

Since each step of DISTILLER may take hours or even days to finish, the result 
of each step is sent to the user by email. This notification email is of a 
standard format, with subject `DISTILLER process result notification email'. On 
the first line, the information about the finished process and the 
corresponding user project is presented, followed by the link to the DISTILLER 
result file. It is followed by the link to the supplementary file bundle 
required for ViTraM visualization.


\subsection{ViTraM}\label{sec:dist-vitram}
For DISTILLER output analysis, we use the module visualization software ViTraM 
(\textit{VIsualization of TRAnscriptional Modules}) to visualize overlapping 
transcriptional coexpression modules together with the motif information in an 
interactive way. In this chapter, we will only briefly discuss this tool. For 
more details we refer to \cite{Sun2009}. 

Note that for this case study, we used ViTraM 2.0 (download at \cite{ViTraM}). 
ViTraM (v2.0) is written in JAVA; hence it can run under any platforms with 
JAVA support. 
On Windows ViTraM comes in two versions: vitram\_Windows\_512M.bat or 
vitram\_Windows\_2G.bat. The former has a minimum memory requirement of 1Gb 
RAM, and we advise to use a computer with at least 3Gb RAM to run 
vitram\_Windows\_2G.bat. % Please refer to \textbf{Note 1} 
%for compatibility issues and memory requirements of the tool.


\subsubsection{ViTraM input file format}
The visualization software ViTraM requires an XML file and an expression file 
as input. The XML file contains the information of the transcriptional modules 
that will be visualized: the genes and conditions composing the modules and 
additional information such as motifs that are assigned to modules by 
DISTILLER. Due to the complexity of the XML format, an extra tool called 
XMLCreator is available at the ViTraM website to automatically generate the XML 
file from the `.m' output file and the supplementary expression data file of 
DISTILLER (see Section \ref{sec:distiller-outfile-format}). 
For the purpose of this study, we will not discuss the XML file format directly 
used by ViTraM. 
Interested users can find more detailed information in \cite{Sun2009}. 
Conveniently, the tool also generates a smaller expression data file containing 
only the relevant expression values from those genes and condition contrasts 
existing in the result modules contained in the `.m' file. Moreover, the tool 
can also incorporate extra information into XML file, for example, the motif 
data contained in the binary supplementary file mentioned in Section 
\ref{sec:distiller-infile-format}. 


%\subsection{Input file formats}\label{sec:dist-format-input}
%
%Different file formats are used by COLOMBOS, DISTILLER and ViTraM. Depending 
%on 
%their functions, the input files are explained in this section and the output 
%files in the next one. Sample files that contain the data used for our case 
%study are also provided (see Section \ref{sec:dist-sample}). 
%
%
%\subsection{Output file formats}
%
%Below, the different output file formats are explained in each subsection.






\subsection{Sample files}\label{sec:dist-sample}

The following four sample files are provided as example input files for this 
case study. The file expdata\_COLOMBOS\_module\_information.txt is an example 
of the exported module data file of COLOMBOS, which, as one of the input 
formats of the expression data for DISTILLER, contains both the gene and 
condition contrast information, and the log ratio expression data. The file 
expdata\_DISTILLER\_Expression.txt is an example of the other expression data
format accepted by DISTILLER. The file binary\_DISTILLER\_Motif.txt is the motif
data containing motif scores for each gene. And finally, the file 
operon\_DISTILLER\_OperonGenes.txt is the example operon information file.

Furthermore, three example output files, corresponding to the output generated
after each step of DISTILLER, are provided. DISTILLER\_Output.m is the output of
the seed module identification step, DISTILLER\_FilteringOutput.m is the output
of the module filtering step, and DISTILLER\_outputExtended.m is the output of
the module extension step. Files expdata\_DISTILLER\_supple.txt and
binary\_DISTILLER\_Motif\_supple.txt are provided as the supplementary files 
used for visualizing modules contained in those sample output files.

Finally, ViTraM\_Modules.XML is provided as the XML file describing the modules,
and ViTraM\_Expression.txt is provided as the expression data file used by
ViTraM.

One can download all the example files from here \cite{DISTILLER-sample} or at 
the DISTILLER website \cite{DISTILLER} (follow the sample file link of the 
second references).




\section{Methods}

We will describe the steps required to use two web services, COLOMBOS and 
DISTILLER, to identify coexpression modules around a (set of) gene(s) of 
interest (query genes). This will be illustrated by applying them to the query 
gene \textit{sodA} as a case study.

The first analysis flow is based on the web service COLOMBOS. We will show how 
COLOMBOS can be used to search for genes that are coexpressed with a (set of) 
query gene(s). In contrast to COLOMBOS, the second approach using DISTILLER 
detects coexpressed gene sets and their regulatory program across the whole 
data set in a single run in an unsupervised way. It searches for gene sets that 
are coexpressed and contain the same motif(s) in their promoter region by 
combining expression and motif data.


\subsection{Identifying coexpression modules containing \textit{sodA} using 
COLOMBOS}\label{sec:dist-module-colombos}

To search for coexpressed genes using COLOMBOS, we will first create an initial 
expression module by specifying a set of genes of interest (query genes), and 
use COLOMBOS to extract the most relevant condition contrasts, where the chosen 
genes are differentially expressed.  
Next, the module is extended with genes that are coexpressed with 
the initial gene set under the preselected set of condition contrasts. Here, we 
explain the workflow based on our case study with a single query gene, 
\textit{sodA}.

\subsubsection{Create the initial module}

\paragraph{Step 1}	Go to COLOMBOS website \cite{COLOMBOS}.

\paragraph{Step 2}	Create the initial module based on the known gene set of 
interest (query genes) and the biological conditions that are known to affect 
these genes when changed.

\begin{adjustwidth}{5mm}{0cm}
\distsubparagraph{Step 2.1}	To start the analysis, click on `data 
analysis' in the title bar (Fig. \ref{fig:colombos}) to bring up the data 
operation interface in the center part of the website. 
It is separated horizontally into two parts. At the left hand side, 
there is a `workspace' information panel that lists all the modules created in 
a tree structure. The workspace panel is always visible when residing on the 
data analysis page. (Note that you can always move away from the data analysis 
page to any of the other COLOMBOS pages, such as the help sections, at any 
time; the state of the data analysis page will be preserved for whenever you 
switch back.)  Since no module has been created, the workspace panel is 
currently empty. The other part of the page, referred as the operating space in 
this text, is where the visualization and analysis of the expression data takes 
place and is currently occupied by the `Data analysis overview' panel 
(hereafter referred to as `overview panel'). This panel serves as an intuitive 
guide showing the user the available data analysis options depending on the 
current state of the workspace along with some brief explanations. The `Main' 
button at the top right of the workspace will always bring up the overview 
panel again in the operating space.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
Since we just started a new session, there is only one button `select organism' 
available. Click it, and select the organism `\textit{Escherichia coli}', the 
one that this case study focuses on, from the dropdown box that appears. Notice 
now this species appears in the workspace panel at left as the root of the 
tree.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
\distsubparagraph{Step 2.2}	After the species is picked, the content of the 
overview panel is also chanced accordingly. The button `select 
organism' is replaced by two other buttons `organism info' and `add module'. 
The workspace panel now also shows that \textit{E. coli} data is loaded. 
Clicking on the \textit{E. coli} entry in the workspace panel has the same 
function as clicking the `organism info' button in the operating space: it will 
bring up detailed information about the organism, its compendium, and 
associated additional data sources. 
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
We will proceed to create our initial module of the gene \textit{sodA}. The 
`Add module overview' panel appears in the operating space after clicking the 
`add module' button. In this panel, there are three buttons `select gene only', 
`select contrasts only', and `select gene and contrast'. As explained on the 
page, each corresponds to a different method for creating a module.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
For this case study, we will only manually specify the gene \textit{sodA} 
without specifying any relevant conditions ourselves.  Instead, we will have 
COLOMBOS automatically identify the most relevant condition contrasts for sodA 
based on its expression values. This function is provided by the leftmost 
button `select gene only'. Clicking it brings up the gene selection option 
panel in the operating space. Four available options are listed here: `By gene 
name/locus tag', `By transcriptional regulation', `By pathway', and `By 
transcription unit'. We want to select a specific gene (\textit{sodA}), so the 
first option is our choice. Click the yellow box surrounding the text `By gene 
name/locus tag', fill in `\textit{sodA}' in the appeared text box (one gene per 
row), and then click the `Done' button located at the bottom to save your 
input. If one or more of the query genes is not recognized or included in the 
compendium, an error message will appear listing the unresolved entries.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
\distsubparagraph{Step 2.3}	As mentioned in previous step, we chose the option 
`select genes only' to create the module. After specifying the genes, a `ranked 
contrast selection' panel (Fig. \ref{fig:colombos-ranking}) will appear in the 
operating space, 
which allows the user to select contrasts ranked by a score that prioritizing 
those show the highest magnitude of change and most coherent coexpression for 
the selected set of genes (see Appendix \ref{apd:contrast-score}). The panel is 
separated into two parts horizontally. 
At the left, the condition contrasts are ranked according to the 
score from highest till the lowest. A cutoff value can be specified in the box 
above the list to select only those contrasts with higher scores. The 
figures to the right are meant to provide additional helpful information and 
show, from left to right, a density plot showing the distribution of the 
contrast scores across the whole compendium and a plot showing the number of 
contrasts that will be added if a given cutoff is selected.
\end{adjustwidth}

\begin{figure}[tb]
	\centering
  	\includegraphics[width=1\textwidth]{Fig4-Condition-contrasts-ranking-panel.png}
	\caption[COLOMBOS contrast ranking interface]{\textbf{The `ranked contrast 
	selection' interface to choose contrasts based on the specified module 
	genes (\textit{sodA} in this particular case).}}
	\label{fig:colombos-ranking}
\end{figure}

\begin{adjustwidth}{5mm}{0cm}
For our initial module, we chose cutoff value 3. There is no absolute guideline 
to choose the cutoff. Several try-outs may be needed to identify the optimal 
one. Input 3 in the box above the list at the left, and click `Ok' button at 
the side. The contrast list is then filtered, and all contrasts left in the 
list are automatically selected to be added to the module. Click `Done' button 
at the bottom, and fill in `sodA\_c3' as the module name to create our initial 
module.
\end{adjustwidth}

The module `sodA\_c3' is now created and will appear in the module tree in the 
workspace at the left side as a node directly under the root node (presenting 
the selected organism \textit{Escherichia coli}).



\subsubsection{Inspect the existing modules}

\paragraph{Step 3}
After a module is created, one can review its information through various 
available options.

\begin{figure}[b]
	\centering
  	\includegraphics[width=1\textwidth]{Fig5-Module-overview.png}
	\caption[`sodA\_c3' module overview]{\textbf{The overview tab of 
	the module `sodA\_c3'.}}
	\label{fig:colombos-init-module}
\end{figure}

\begin{adjustwidth}{5mm}{0cm}
\distsubparagraph{Step 3.1}	Click the module you want to check in the workspace 
at the left to show its module overview tab (Fig. 
\ref{fig:colombos-init-module}) in the operating 
space at right. 
It provides a brief summery of the current module, and serves as a 
helping guide to analyze and visualize the module's composition. For each 
module, the summary section shows its name followed by its description at the 
top. Below the number of included genes and contrasts are indicated, as well a 
list of Gene Ontology (GO) enrichment scores (see Appendix 
\ref{apd:enrichment}) for the 
genes in the module. Only the GO terms of the class biological process are 
calculated for their enrichment scores, and only those with $p$-values $< 0.1$ 
are shown. 
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
At the bottom of this panel, there are four buttons to \textit{visualize}, 
\textit{edit}, \textit{split}, or \textit{delete} the module. The function of 
\textit{visualization} and \textit{delete} is straightforward. The 
\textit{edit} function allows user to modify either the genes or 
the contrasts of the module. Instead, the \textit{split} function breaks the 
current module into several new ones by separating either its genes or its 
contrasts into distinctive groups and creating a module for each selected group.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
\distsubparagraph{Step 3.2}	Each module can be visually presented as a 2D 
heatmap. 
In the heatmap, each square represents one gene's expression log ratio for a 
certain condition contrast. A red color indicates over-expression while a green 
one shows under-expression. The intensity of the color corresponds to the 
magnitude of the change. The brighter the color, the higher the absolute ratio 
value.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
To view the heatmap, click `Visualization' button in a module's overview tab to 
switch to heatmap tab. In the tab, the heatmap is shown at the center with an 
information panel at the right hand side. Hovering over the heatmap, the 
information of the gene and the condition contrast under the cursor will be 
shown in that information panel. Various options exist to group the condition 
contrasts in the heatmap according to their experiments, condition properties, 
and the condition ontology terms that their associated condition properties 
belong to. It can be selected from the dropdown box located at the left corner 
of the bar below the heatmap. The heatmap of our module sorted on condition 
ontology (default option) is shown in Figure \ref{fig:colombos-heatmap-m1}.
\end{adjustwidth}

\begin{figure}[tb]
	\centering
  	\includegraphics[width=1\textwidth]{Fig6-Heatmap-module-1.png}
	\caption[Heatmap of module `sodA\_c3']{\textbf{The visualization tab of 
	module `sodA\_c3' showing the heatmap.}
	In this heatmap, the contrasts are grouped by their ontology annotations. 
	The contrast groups belonging to the ontology term `response to oxygen 
	level' are marked in the figure.}
	\label{fig:colombos-heatmap-m1}
\end{figure}

Checking our module `sodA\_c3', it shows that the top 53 condition contrasts, 
where \textit{sodA} is most differentially expressed, have been selected. Among 
those, 34 contrasts belong to the condition ontology term `response to oxygen 
levels'.  This ontology term includes condition properties that are linked to 
cellular processes that are dependent on oxygen availability, such as fnr 
mutations (a global oxygen responsive transcriptional regulator), NO$_2$ 
(electron transport decoupler), agitation of the growth medium, actual oxygen 
levels, etc. Since the function of \textit{sodA} itself is linked to processes 
related to oxygen availability, one can expect its expression level to change 
prominently under these 34 contrasts, and indeed COLOMBOS successfully 
identifies them as the most relevant contrasts of \textit{sodA}. There are also 
17 contrasts belong to ontology term `growth', with three in common with the 34 
contrasts mentioned before. The ontology term `growth' is very general, 
grouping experimental conditions that trigger various biological processes 
simultaneously at a global cellular scale. Based on this, it is not unexpected 
to find \textit{sodA} differentially expressed under contrasts belonging to 
this ontology term.


\subsubsection{Extend the genes of a module}

\paragraph{Step 4} To extend the module with additional genes, we will use the 
edit function. 

\begin{adjustwidth}{5mm}{0cm}
Click `edit module' button to bring the `Edit' tab in front, 
where three options are available: edit the name/description, the genes, or the 
contrasts. Click the `edit genes' to modify the genes of this module. A gene 
modification panel appears in the `Edit' tab with all options to edit genes. 
The options in green boxes indicate ways to add genes to the module, whereas 
those in red boxes indicate ways to remove genes. We will not explain all the 
options here. Interested users please refer the tutorial and/or the help 
sections of the website for detailed explanations.
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
For this case study, we will use the last option `Add new genes based on 
expression'. This will allow the user to add additional genes, whose expression 
patterns are most (anti-)correlated to the expression pattern of the module. 
Click the green box surrounding the text of this option; a `ranked gene 
selection' panel will appear in the `Edit' tab. 
\end{adjustwidth}

\begin{figure}[b]
	\centering
  	\includegraphics[width=1\textwidth]{Fig7-Gene-extending-panel.png}
	\caption[COLOMBOS `ranked gene selection' interface]{
	\textbf{The `ranked gene selection' interface to extend module `sodA\_c3' 
	with the most relevant genes.} 
	The red circle indicates \textit{sodA} in the module. As it's expression 
	profile is perfectly correlated with itself (the correlation score of 1), 
	the circle is located at point (1, 0).}
	\label{fig:colombos-gene-extend}
\end{figure}

\begin{adjustwidth}{5mm}{0cm}
This panel (Fig. \ref{fig:colombos-gene-extend}) functions in a similar way to 
the `ranked contrast selection' panel explained in Step 2.3. 
On the left side, a gene list ranked by the degree of the correlation of each 
gene's expression profile with the mean profile of the current module's gene 
set under the selected condition contrasts of this module. 
The higher the score, the more similar the profiles are. 
A cutoff value can be specified in the box above the list to select only those 
genes with higher scores. 
Based on the \textit{(uncentered) correlation}, there are three different 
types of correlation scores available to rank the genes. 
Here we will use the default `correlation', which calculates the 
\textit{Uncentered Pearson Correlation} as the score. 
Other options can be chosen from the dropdown box that will appear by clicking 
the text `correlation' in the bar on top of the ranked gene list. 
Please refer Appendix \ref{apd:gene-score} for more information 
about these options and how to calculate them. 
Also in the panel, the figures to the right are meant to provide additional 
helpful information and show, from left to right, 
a density plot showing the distribution of the 
correlation scores across all genes in the compendium and a plot showing the 
number of genes that will be added if a given cutoff is selected. The circles 
connected by a red vertical line in the density plot indicate the gene(s) that 
already exists in the current module and its correlation score. 
\end{adjustwidth}

\begin{adjustwidth}{5mm}{0cm}
To extend our module, we will use a cutoff value of 0.75. Input 0.75 in the box 
above the gene list at the left, and click `Ok' button at the side. The list is 
then filtered, and all genes to be added are automatically selected (checked) 
in the list to the left. With the option `Create new module upon edit' (located 
below the gene list), the original module is kept unchanged, and the extended 
module is saved as a new module. Chose this option, then click the `Add' button 
at left, and specify `sodA\_c3\_g.75' (in the popup window that appears) as the 
name of the new module to create it. It will now appear as a node below the 
original module `sodA\_c3' in the workspace. 
\end{adjustwidth}

The resulting module now contains 35 genes with an expression profile similar 
to that of \textit{sodA} under the selected contrasts (see heatmap, Fig. 
\ref{fig:colombos-heatmap-final}). 
Amongst the genes in the module are \textit{cyoA}, \textit{cyoB}, 
\textit{cyoC}, \textit{cyoE}, subunits of the cytochrome b terminal oxidase 
complex involved in aerobic respiration \cite{Cotter1992}, \textit{sdhA}, 
\textit{sdhC}, \textit{sdhD}, encoding the succinate dehydrogenase (SdhCDAB) 
active during Krebs cycle catalyzing the oxidation of succinate to fumarate 
under aerobic conditions \cite{Wilde1986}, and \textit{sucA}, \textit{sucB}, 
\textit{sucC}, \textit{sucD} involved in generating succinyl-CoA, one of the 
reactants in the Krebs cycle \cite{Buck1989}. 
Considering the functions of these gene products (see also enriched GO terms in 
Table \ref{tab:GOenrich}), it is not at all unexpected that their expression 
levels are influenced by oxygen availability. 
Moreover, many genes in module `sodA\_c3\_g.75' also share common regulators 
with \textit{sodA}, such as ArcA, Fur, FNR, CRP, etc. Hence, \textit{sodA} 
(encoding a superoxide dismutase activity) being coexpressed with the genes 
involved in aerobic respiration might be essential to protect the cell against 
oxidative stress.

\begin{figure}[tb]
	\centering
  	\includegraphics[width=1\textwidth]{Fig8-Heatmap-final-module.png}
	\caption[COLOMBOS heatmap of module `sodA\_c3\_g.75']{
	\textbf{The heatmap of AU2 module `sodA\_c3\_g.75'.}
	The row of expression data that corresponds to \textit{sodA} is
	marked by the arrow. The expression data of the gene groups that are 
	prominently coexpressed with \textit{sodA}, namely cyoABCD, sdhACD and 
	sucBCD, are marked by the named boxes.}
	\label{fig:colombos-heatmap-final}
\end{figure}


\subsubsection{Summary}

The fact that we obtained module `sodA\_c3\_g.75' clearly show the strength of 
the simple and straightforward method implemented in COLOMBOS. Although this 
COLOMBOS coexpression module does not necessarily imply coregulation of its 
genes, this could be investigated further (see Fig. 
\ref{fig:workflow-distiller-colombos}). 
The promoter regions of the genes in this module could now be used as input for 
motif detection methods. This task could be achieved either by screening the 
promoter regions with Position Specific Scoring Matrices (PSSMs) of previously 
characterized motifs, or by using \textit{de novo} motif detection methods 
\cite{Tompa2005, Storms2010}. 
Note that some of the genes in it are unannotated, the COLOMBOS web service can 
also serve as a tool to help biologists identify interesting future research 
candidates.




\subsection{Identifying transcriptional modules containing \textit{sodA} with 
DISTILLER}

As described above, DISTILLER \cite{Lemmens2009} is a data integration 
framework that can 
automatically identify condition dependent transcriptional modules by combining 
expression data with information on the interaction between a regulator and its 
corresponding target genes (here through motif data). Here, we will show how to 
setup the related parameters and run each of the three steps of DISTILLER 
through our web service \cite{DISTILLER}. 
After transcriptional modules are obtained using DISTILLER, 
the software ViTraM \cite{Sun2009} is used to visualize overlapping 
transcriptional coexpression modules together with the motif information in an 
interactive way. At the end, we briefly discuss those modules containing 
\textit{sodA} in DISTILLER results.

\subsubsection{Part.1 Running DISTILLER algorithm}\label{sec:dist-distmodule}
Every search for regulatory modules targeting one specific gene or a set of 
genes starts from the expression and motif data of all genes of the species. 
DISTILLER will then identify all possible modules. From the resulting modules, 
the user can filter out those involving his or her query genes. As a case 
study, we will use DISTILLER to identify the regulatory modules containing 
\textit{sodA}.

To reduce the running time of DISTILLER, the motif data can be reduced by 
removing all the motifs that are irrelevant to the specific query gene set.  
Here, we omitted motifs that were not present in the \textit{sodA} promoter 
region from our motif dataset. Only 7 motifs (ArcA, Rob, SoxS, Fur, IHF, 
FNR, MarA, and CRP) remain.


\paragraph{Create a user profile and a project}
DISTILLER algorithm is computationally expensive, as it explores the complete 
search space. Depending on the specified parameters and the dataset size, Each 
step could take hours or days to finish.  Hence, a user account is required 
with minimally user's email information, and a project should be defined for 
each dataset.  In this manner, user can simply setup the system to run each 
step of the algorithm and leave the site.  
When process finished, user is notified by an email (see Section 
\ref{sec:distiller-email} for its format and content).
He can then log in and select the specific project to check the results or 
continue with next step.

%Since DISTILLER can be time consuming, a user account needs to be created and 
%a project should be de?ned for each dataset.

\vspace{-5mm}\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
% command required to use \subparagraph inside adjustwidth env.
% ref: 
%http://tex.stackexchange.com/questions/60992/strange-compilation-problem-using-adjustwidth-environment-from-changepage-packag
\subparagraph{Step 1}	Go to the DISTILLER web service \cite{DISTILLER}. An 
existing user can login through the front page. 
New users can create a user profile by clicking the `New User' 
button and completing the required information. Subsequently, click the `Create 
User' button. Note that the user's email address is required for sending an 
email notification of the results (see Section \ref{sec:distiller-email}).

\subparagraph{Step 2}	After log in, the project selection interface will 
appear, where one can directly select an existing project from the combo box to 
continue, or create a new one by clicking `Create New Project' and giving a 
name and a brief description of the project. Here, we create a new project 
named `Ecoli\_sodA'.

\subparagraph{Step 3}	After choosing or creating a project, the DISTILLER 
panel will appear where the user can run the three main steps of DISTILLER, 
each of which has its own separate tab. As our project is a new one, the user 
will see the `Module Detection' tab (Fig. \ref{fig:distiller}), while the 
`Module Filtering' and `Module Extension' tabs are disabled.
\end{adjustwidth}


\paragraph{Identifying seed modules} This step allows the user to identify 
seed modules from the source data.


\vspace{-5mm}\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\subparagraph{Step 4} For a new project, user need to first upload the required 
expression data file and motif data file (see Section 
\ref{sec:distiller-infile-format}) to be analyzed by DISTILLER program, and 
preprocess them to check the format 
and integrate of the files.
As the system support different expression data file formats and various 
compression types, it is user's responsibility to specify the correct 
information by selecting the right options for the `Compression' and `Data file 
format' items of the `Expression Data Parameters' respectively.

Click the `Upload Expression Data File' button to upload the expression data 
file, and the `Upload Binary Data File' button for the motif data file. Here we 
will use sample files expdata\_DISTILLER\_Expression.txt and 
binary\_DISTILLER\_Motif.txt respectively, and specify `Flat file format' as 
`Data file format' and `No' for `Compression'. 
Click `Preprocessing' button to do the integrate check and the required file 
conversion. As this step can take time to finish, user can now leave the site, 
and wait for the notification email to continue.

%For a new project, no source data files have 
%been uploaded, hence only the `Load file' button at the lower right corner is 
%enabled, and the `run Distiller' button is disabled (Fig. 
%\ref{fig:distiller}). 
%Users must first upload the expression data file (use sample file 
%expdata\_DISTILLER\_Expression.txt) and the motif data file 
%(binary\_DISTILLER\_Motif.txt). 
%Upload the required data files by clicking the `Load File' button. Note that 
%the web service automatically recognizes the data type of a file according to 
%the prefix of its name. 
%Please refer Note 10 for the mandatory file name prefix of each input file 
%type.

\subparagraph{Step 5} When receiving the notification email of the 
preprocessing step, the user can go back to the website to proceed with the 
main steps.  
Open the DISTILLER website, log in with your user account, and select the 
previously created project. The same interface as seen in Step 4 appears.
There are three groups of parameters need to be specified for this step: the 
binary data parameters related to the 
motif data (see Step 5.1), the expression data parameters (see Step 5.2), and 
the general parameters (see Step 5.3).

\vspace{-5mm}\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\subparagraph{Step 5.1} Binary data parameters: 1) Binary Support specifies 
the minimal number of motifs that the genes in a module should have in common; 
2) Binary Thresholds is the minimal score a motif instance in the promoter 
region of a gene should have to be considered as present in that promoter 
region. 
It is applied on the motif data to convert its \textit{continuous values} into 
the \textit{binary ones}. 
Any value which is equal or higher than the threshold will be converted into 
value 1, and 0 otherwise, resulting in a binary matrix. 
This binary matrix is then used by the algorithm to identify modules.

For this case study, we choose 1 for the Binary Support and 
0.999 as the Binary Threshold. Hence the genes belonging to a module must share 
at least 1 motif and a gene is considered to have a specific motif if the 
corresponding value in the motif data is equal or higher than 0.999.

\subparagraph{Step 5.2}	Expression data parameters: 1) Box Support specifies 
the minimal number of condition contrasts under which the genes in a module 
should be coexpressed; 2) Box P-value is used to generate the threshold 
bandwidth sequence that is needed to test whether a module passes the 
constraints on the expression data, i.e. whether the genes in the module 
are sufficiently coexpressed within the selected contrasts \cite{Lemmens2009}. 
Here, 100 is chosen for the Box Support and 0.0001 for the Box P-values.

\subparagraph{Step 5.3} General parameters: 1) Minimal Module Size specifies 
the minimal number of genes that should be in a module; 2) Number of 
Randomizations specifies the number of random modules that will be generated 
for computing a threshold bandwidth sequence for the condition contrast 
selection. 
Here, we specify the Minimal Module Size as 4, and the Number of Randomizations 
as 10000. The algorithm uses these parameters to generate 10000 random modules 
consisting of 4 genes to compute the threshold bandwidth sequence.
\end{adjustwidth}

\subparagraph{Step 6} When all parameters are specified, click `Run 
Distiller'. DISTILLER will now enumerate all initial modules, each of which 
contain at least 4 genes coexpressed in at least 100 condition contrasts and 
sharing at least one motif. Note these contrasts are selected based on 
comparing the ordered contrasts bandwidth sequences of given genes with the 
threshold bandwidth sequence (see \cite{Lemmens2009} for details). 
\end{adjustwidth}

When finished, user will receive a notification email (see Section 
\ref{sec:distiller-email}) containing the link to the output file with the
identified modules, and the link to the supplementary file bundle that is 
required to visualize modules using ViTraM.


\paragraph{Filter the raw DISTILLER output} After receiving the notification 
email for the module detection step, the user can go back to the saved project 
in DISTILLER website to proceed with the other steps. 
Now, the tabs `Module Filtering' and the tab `Module Extending' are enabled. 
Note that the user can directly proceed to the module extending step, but due 
to our discussion in Section \ref{sec:dist-distiller}, performing a filtering 
step before extending the obtained modules is highly recommended.

\vspace{-5mm}\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\subparagraph{Step 7} Click the `Filtering' tab. Specify the number of 
modules to be filtered out in this step. Click on `Filter Result' to proceed.
In our case study, we will choose 20 modules to obtain the top 20 modules 
ranked by their scores computed by DISTILLER (see Section 
\ref{sec:dist-distiller}). Similarly, the result of the process will be 
notified by email.
\end{adjustwidth}


\paragraph{Extend DISTILLER seed modules} When receiving the notification email 
for the filtering step, the user can now go back to the DISTILLER 
website to proceed with the `Module Extension' step. 
In this step, extra genes are incorporated into the modules obtained if they 
comply with the relaxed criteria.

\vspace{-5mm}\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\subparagraph{Step 8} Two parameters need to be specified to run this 
step. Candidate genes have to comply with both parameters in order to be 
included in the extended module. The only exception is made for the genes 
belong to an \textit{operon}, which will be explained in Step 9. 

\vspace{-5mm}\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\subparagraph{Step 8.1} Extended Motif Threshold: this is the same type of 
parameter as the Binary Thresholds specified in Step 5.1. 
By choosing a less stringent (lower) value for the Extended Motif Threshold 
than was chosen for the Binary Thresholds, a gene will have more motif 
candidates. As a result, more genes can satisfy the Binary Support parameter. 
In the case study, we use 0.95 for the Extended Motif Threshold (as compared to 
0.999 for the Binary Thresholds in Step 5.1).

\subparagraph{Step 8.2}	Correlation Percentage: a correlation threshold applied 
to select candidate genes for module extension. 
Given a Correlation Percentage of $p$, only those genes whose expression 
profile correlation, calculated based on the mean expression profile of a 
module, is higher than $p$, will be considered as the candidates to extend that 
module. Here we choose $p$ as 0.95.
\end{adjustwidth}

\subparagraph{Step 9} Optionally, the genes of a module can be further 
extended with \textit{operon information} if available. 
Candidate extension genes belonging to an operon whose first gene is present in 
a seed module, only need to satisfy the criterion for the expression profile 
(Correlation Percentage parameter) to be included into the module. 
The operon information (i.e. which genes belong to which operon) is included in 
the analysis by uploading an operon information file (see Section 
\ref{sec:distiller-infile-format}) with the prefix `operon\_' in its file name.

\subparagraph{Step 10} After specifying the parameters and optionally 
uploading an operon data file, the user can click `Extend Modules' to run the 
module extending step. 
A notification email is sent when the process is finished.
\end{adjustwidth}

%Note that the DISTILLER algorithm is a global data exploration method that 
%identifies all modules that meet user specified criteria.

As a global data exploration method, the DISTILLER result modules obtained 
contains those irrelevant to our target gene \textit{sodA}. 
We simply sift out the modules that contain gene \textit{sodA} to continue.




%After receiving the final output file of DISTILLER, the user can then 
%specifically select the modules that contain his or her query genes, in our 
%case the gene sodA. Note that here we only discussed how to use DISTILLER to 
%identify transcription networks. However, a versatile tool like DISTILLER has 
%many other application domains. Please see \textbf{Note 12} for more details. 


\subsubsection{Part.2 Visualization of DISTILLER modules with ViTraM}
Here we show how to use ViTraM to analyse modules discovered by DISTILLER. 
First, as mentioned in Section \ref{sec:dist-vitram}, the DISTILLER output files
need to be converted into ViTraM compatible format. 
Then, the generated files are used in ViTraM to visualize the obtained modules. 
We will discuss both steps separately.

Note that due to the randomization, DISTILLER is non-deterministic. Thus, it is 
possible that when running DISTILLER in exactly the same way as outlined here, 
the recovered modules are different from the ones obtained in this example (as 
provided in `DISTILLER\_outputExtended.m' sample file, see Section 
\ref{sec:dist-sample}). 
Use the corresponding sample file to proceed with exactly the same 
results.


\paragraph{Prepare ViTraM readable files}

\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\vspace{-5mm}\subparagraph{Step 1} Download the XMLCreator from the ViTraM 
website\cite{ViTraM} and unzip the file. Run the XMLCreator by typing the 
command `java -jar -Xms256m -Xmx512m XMLCreator.jar' in a terminal. Windows 
users can click file `XMLCreator\_Windows\_512M.bat' to run the XMLCreator. 

\subparagraph{Step 2} In the popup window, choose `DISTILLER' as module 
detection tool and click on `OK'. 
A new window appears where user can specify the input and output files. 
The required input files are the DISTILLER output data file of `.m' format and 
the `Expression Data' file (see Section \ref{sec:dist-vitram}). 
In addition, other files can provide extra information for each module that can 
be visualized along with the modules by ViTraM. 
Some examples of such extra information are the motif information of each gene, 
the functional annotations of the genes, or the information on what was 
measured in each experimental conditions, etc.
Finally, the names of the output files need to be specified. Two output files 
will be generated: the module XML file and the corresponding expression data 
file consisting of the expression values for those genes and condition 
contrasts present in any of the modules. 
When ready, click the `Make XML File' button to generate these output files. If 
files are generated successfully, a window will appear to notify user.
\end{adjustwidth}

As an example, we will use sample files provided with this chapter to show how 
to run XMLCreator. First, specify the file `DISTILLER\_outputExtended.m' as the 
`DISTILLER output data', and the file `expdata\_DISTILLER\_supple.txt' as the 
`Expression Data'. This is the minimal input that is required to run 
XMLCreator.
In addition, we also provide the file `binary\_DISTILLER\_Motif\_supple.txt' as 
the `Motif Data' providing extra motif information to be visualized together 
with the modules.
Specify `ViTraM\_Modules.XML' and `ViTraM\_Expression.txt' as the output file 
names in data field `XML Data' and `Expression Data'. Click `Make XML File' 
should generate the output file successfully.
To run XMLCreator on the DISTILLER output, pleases always use the supplement 
files specified in the same notification email of an `.m' output file (see 
Section \ref{sec:distiller-email}).


\paragraph{Run ViTraM}

From the ViTraM website \cite{ViTraM}, the user can 
download ViTraM v2.0 and unzip the file. Note, a registration is required by 
providing us some basic information, and the user must accept our software 
license.

\begin{adjustwidth}{3mm}{0cm}
\nointerlineskip\leavevmode
\subparagraph{Step 3} To run ViTraM under windows, click on 
`ViTraM\_Windows\_512M.bat' or `ViTraM\_Windows\_2G.bat' provided.
The latter can handle larger datasets, but does require more than 2Gb physical 
memory in the computer. To run ViTraM under linux or mac, call 
`ViTraM\_Linux\&Mac.sh' file.

\subparagraph{Step 4} Load `ViTraM\_Modules.XML' and 
`ViTraM\_Expression.txt' generated in Step 2 by clicking on `Open Module XML 
File' and `Open Expression File', respectively. 
After the data are loaded, they can be visualized by clicking on `Load All 
Modules' in the `Module Selection' panel. 
This extra loading step enables visualizing a subset of modules in the large 
dataset, where directly visualizing all the modules in it after data is loaded 
generally causes memory issues. 
In such a case, user can use `Filter' panel first to select only the relevant 
modules, then to visualize them. This will significantly reduce the loading 
time to visualize modules.

\subparagraph{Step 5} Subsequently clicking on `View Modules' in the 
`Module Display' panel to show the modules in the main window `Module Display'. 
In this window, the genes are listed at the left, and the conditions are listed 
at the top. Each module is represented as one or a group of boxes with 
its id shown at the top left corner of each box. Note, each module uses a 
distinctive color for its boxes.

\subparagraph{Step 6} Click `View Gene Properties' in `Gene Properties 
Display' panel, the 
genes' extra properties can be visualized together with the modules in an extra 
window located at the left side of the `Module Display' window. Since we 
provided motif data while generating XML file in Step 2 of this section, they 
will now appear in this window. The color of each cubic represents the value of 
the motif score explained in Section \ref{sec:dist-vitram}. 
The red color corresponds to the high value and the green the low one. If a 
score is higher than a threshold, a cross will appear in the corresponding 
cubic. For more information about this panel, we refer user to the manual of 
ViTraM.

\subparagraph{Step 7} The `View Experiment Properties' in the `Experiment 
Properties Display' 
panel can open up one extra window that shows the relevant condition data if 
loaded. In this case study, no such data is available.

\subparagraph{Step 8} By clicking on `Favorite Genes' in the `Filter' panel 
at the right hand side, it is possible to display only those modules containing 
a particular gene. 
The pop-up window shows on the left a list of all genes. Select the gene 
`b3908', corresponding to \textit{sodA}, and click the arrow button 
`\textrightarrow' to 
add it to the favorite gene list at the right. When clicking `OK', only modules 
containing the \textit{sodA} gene will be displayed.

\subparagraph{Step 9} To optimize the visualization of modules that 
overlap, click on `Automatic ordering' in the `Module ordering' panel and 
subsequently `Run Overlap Index' to layout the currently displayed modules in 
the most optimal way (for details on the algorithms that identify the optimal 
 display of overlapping modules, we refer to \cite{Sun2009}).

\subparagraph{Step 10} Click on `View Modules' and subsequently `Refresh 
Modules' in the `Overview \& Heatmap Display'. Then click on `Adding Heatmap'. 
The expression values of the genes for the condition contrasts in the currently 
displayed modules will be shown by means of a heatmap.
\end{adjustwidth}


\subsubsection{Analyzing resulting DISTILLER modules containing \textit{sodA}}

In the case study we used DISTILLER to search for modules of coexpressed
genes sharing at least one motif instances for the same regulator.
Resulting modules including the motif instances shared among genes
within each module are listed in Table \ref{tab:distModules}. 
Since we are interested in \textit{sodA}, transcriptional modules containing 
this gene were selected and visualized using ViTraM. 
They are modules 2, 3, 8, and 12 (Fig. \ref{fig:vitram}). 
The numbering of the modules corresponds to their ranks in the filtered
results. Note that DISTILLER works as a global method and thus
identifies all possible regulatory modules in the dataset (so not only
those with \textit{sodA}). This explains why the numbering of the modules
containing \textit{sodA} is not consecutive. The lower the number, the more
significant a module is.

\begin{figure}[b]
	\centering
  	\includegraphics[width=1\textwidth]{Fig9-Distiller-modules-BW.pdf}
	\caption[DISTILLER \textit{sodA} related modules visualized in ViTraM] 
	{\textbf{DISTILLER \textit{sodA} related modules visualized in ViTraM.}
	The figure shows the four modules containing \textit{sodA}. Each module is 
	indicated with one or more squares and expression values are indicated by 
	heatmap. Module 2 is indicated by dotted lines, module 3 by solid line, 
	module 8 by dash-dotted lines, and module 12 by dashed lines.}
	\label{fig:vitram}
\end{figure}

\begin{table}[tb]
	\caption{Overview of the 20 transcriptional modules identified by DISTILLER}
	\label{tab:distModules}
	\begin{small}
	\begin{tabular}{p{5mm} p{2cm} c c p{4.5cm}}
	\toprule
	{\bf ID} & {\bf Motifs} & {\bf \# Genes} & {\bf \# Conds} & {\bf Genes in 
	\textit{sodA} modules} \\
	\midrule
	1 & Fur	& 22 & 97 & \\
	2 &	MarA, SoxS & 4 & 77	& \textit{fpr}, \textit{poxB}, 	
	\textit{\textbf{sodA}}, \textit{zwf} \\
	3 &	ArcA, CRP &	7 &	79 & \textit{acnA}, \textit{acnB}, \textit{aldA}, 
	\textit{gltA}, \textit{osmY}, \textit{sdhC}, \textit{\textbf{sodA}} \\
	4 &	FNR, IHF & 4 & 226 & \\
	5 &	ArcA, FNR &	4 &	123	& \\	
	6 &	CRP	& 308 &	107	& \\
	7 &	CRP, FNR & 4 & 87 & \\
	8 &	Fur, CRP & 4 & 146 & \textit{cyoA}, \textit{nupC}, \textit{sdhC}, 
	\textit{\textbf{sodA}} \\
	9 &	SoxS & 4 &	181	& \\
	10 & Fur & 18 & 86	& \\
	11 & MarA &	4 &	206	& \\
	12 & CRP, FNR & 5 &	90 & \textit{aldA}, \textit{cyoA}, 
	\textit{malP}, \textit{pdhR}, \textit{\textbf{sodA}} \\
	13 & IHF &	8 &	87	& \\
	14 & IHF &	53 & 123	& \\
	15 & FNR &	38 & 102	& \\
	16 & CRP &	76 & 125	& \\
	17 & ArcA, CRP &  4 &	148	& \\
	18 & SoxS &	4 &	95	& \\
	19 & Fur & 4 &	100	& \\
	20 & Fur & 22 &	76 & \\
	\bottomrule
	\end{tabular}
	\end{small}
\end{table}

Module 2 contains the genes regulated by both SoxS and MarA. The
identified condition contrasts in this module belong to the ontology
terms `carbohydrate metabolic process', `growth', and `response to
oxidative stress'. SoxS and MarA are known to participate in the removal
of superoxide and nitric oxide and protection from organic solvents
\cite{Touati2000}. 
One can expect that the condition contrasts belonging to `response
to oxidative stress' to be suitable for inducing their expression and
therefore the expression of genes regulated by them. We already
highlighted the link between the removal of superoxide species and
metabolic pathways. It explains why the condition contrasts under
`carbohydrate metabolic process' are found in this module. As discussed
after Step 3 of Section \ref{sec:dist-module-colombos}, to observe condition 
contrasts belonging to `growth' is not unreasonable in this case. 
In addition to \textit{sodA}, the other genes found in this module are 
\textit{fpr} (Ferredoxin-NADP reductase), \textit{poxB} (Pyruvate 
dehydrogenase), and \textit{zwf} (Glucose-6-phosphate 1-dehydrogenase). The 
enriched GO terms of this gene set (Table \ref{tab:GOenrich}) are highly 
consistent with the observed condition ontology terms.

\begin{table}[tb]
\begin{threeparttable}
	\caption{GO enrichment of \textit{sodA} modules\tnote{1}}
	\label{tab:GOenrich}
	\begin{small}
	\begin{tabular}{p{2.2cm} c c p{5.3cm}}
	\toprule
	{\bf Module} & {\bf P-value} & {\bf GO ID} & {\bf GO Term Description} \\
	\midrule
	{\bf\it COLOMBOS} \\
	\hline
	\textit{sodA\_c3\_g.75}	& 2.41e-12 & GO:0006091 & generation of precursor 
	metabolites and energy \\
	& 9.23e-02 & GO:0006800 & oxygen and reactive oxygen species metabolic 
		process \\
	& 3.17e-02 & GO:0006869 & lipid transport \\
	& 2.74e-07&	GO:0022900&	electron transport chain \\
	& 4.22e-02&	GO:0042592&	homeostatic process \\	
	& 1.85e-02&	GO:0045454&	cell redox homeostasis \\[2ex]
	{\bf\it DISTILLER} \\
	\hline
	\textit{module2} & 7.07e-02&	GO:0005996&	monosaccharide metabolic 
	process \\
	& 7.11e-03&	GO:0006800&	oxygen and reactive oxygen species metabolic 
		process \\
	\hline
	\textit{module3} & 3.11e-05&	GO:0006091& generation of precursor 
	metabolites 
	and energy \\
	& 1.77e-02&	GO:0006800&	oxygen and reactive oxygen species metabolic 
	process \\
	& 7.11e-02&	GO:0044262&	cellular carbohydrate metabolic process \\
	\hline
	\textit{module8} & 4.09e-03&	GO:0006091&	generation of precursor 
	metabolites 
	and energy \\
	& 1.06e-02&	GO:0006800&	oxygen and reactive oxygen species metabolic 
	process \\
	& 4.09e-03&	GO:0022900&	electron transport chain \\
	\hline
	\textit{module12}& 1.42e-02& GO:0006800& oxygen and reactive oxygen 
	species metabolic process \\
	\bottomrule
	\end{tabular}
	\begin{tablenotes}
	\item[1] Enriched GO terms are ordered by GO access ID.
	\end{tablenotes}
	\end{small}
\end{threeparttable}
\end{table}

In module 3, genes are regulated by both ArcA and CRP. Its condition
contrasts are highly related to the ontology terms `growth' and
`response to oxygen levels'. CRP is a global regulator involves in the
degradation of any non-glucose carbon sources and also an antagonist of
catabolite repression. On the other hand, ArcA participates specifically
in a signal transduction system sensing particular aerobic and anaerobic
growth conditions \cite{Compan1994}. 
The functions of the genes found in this module
(\textit{acnA}, \textit{acnB}, \textit{aldA}, \textit{gltA}, \textit{osmY}, 
\textit{sdhC}, and \textit{sodA}) are an intersection
between global metabolic pathways and specific processes responding to
oxygen level changes (see Table \ref{tab:GOenrich}).

Genes in module 8 are both regulated by CRP and Fur. As previously
mentioned, CRP is a global regulator that facilitates bacterial fitness
in function of the availability of different carbon sources. Fur is a
sensor of intercellular iron concentration, and also participates in the
response to reactive nitrogen species \cite{Mukhopadhyay2004}. 
The condition contrasts of
this module mainly involves `carbohydrate metabolic process', `growth',
`detoxification of nitrogen compound', `lactose catabolic
process'. Except for \textit{sodA}, the genes found in this module are 
\textit{cyoA} (Ubiquinol oxidase subunit 2), \textit{nupC} (Nucleoside permease 
nupC), and \textit{sdhC} (Succinate dehydrogenase cytochrome b$_{556}$ subunit).
Altogether the condition contrasts and the genes selected in this module 
reflect the linkage between basic metabolism changes induced by external 
C-source availability (e.g. glucose concentration) and intra-cellular energy
production through the electron transfer chain (see also enriched GO
terms in Table \ref{tab:GOenrich}).  

In module 12, genes are regulated by CRP and FNR,
both master-regulators. FNR activates genes involved in anaerobic
metabolism and represses genes involved in aerobic metabolism. The
enriched GO term (only `oxygen and reactive oxygen species metabolic
process') for this module's gene set reflect this. The selected
condition contrasts of this module constitute a very broad set of
ontology terms, including `carbohydrate metabolic process', `growth',
`dna repair', `sos response', `lactose catabolic process', and
`detoxification of nitrogen compound'. This is in line with the global
regulation exerted by CRP and FNR.


\section{Discussion and Conclusion}

The COLOMBOS web service provides very straightforward and intuitive
ways for analyzing and exploring large-scale expression compendia. It is
intended for specific queries based on users pre-knowledge on genes or
conditions of interest. In each step, this knowledge and user
involvement is crucial to explore various functionalities in order to
generate good quality results. It is a deterministic approach, which
means that when executing exactly the same operations, COLOMBOS will
generate exactly the same result. Furthermore, the analysis methods
implemented in COLOMBOS are fast, generally taking only seconds or less
to generate results for each step.

DISTILLER, on the other hand, is a semi-automatic method. Except for a
limited set of parameters, it requires very little user 
involvement. Guided by both motif information and expression data, the
method tries to compose a global regulatory network that covers the
whole input dataset (see Table \ref{tab:distModules}). 
It is non-deterministic due to the
randomization involved in generating a threshold bandwidth sequence for
the condition selection (see Step 8 in Section \ref{sec:dist-distmodule}).

In our case study we used COLOMBOS and DISTILLER to gain insights in the
functional processes, in which \textit{sodA} is involved, and the regulatory
programs that coordinate them. This resulted in one module indentified
by COLOMBOS and four modules identified by DISTILLER. As discussed in
previous sections, each module does reflect relevant biological
processes, in which genes including \textit{sodA} participate. This does not
necessarily require the modules to overlap, as each might represent
different processes involving different genes. When these two approaches
are used to address the same biological question as was done here, the
distinct features of each approach leads to different but meaningful and
thus complementary results.  

The module `sodA\_c3\_g.75' identified by COLOMBOS contains several genes 
involved in various biological processes
(see Fig. \ref{fig:colombos-heatmap-m1} and the discussion at the end of 
Section \ref{sec:dist-module-colombos}) with very high
and coherent expression values for most condition contrasts (in both
cases where up-regulation or down-regulation occurred). In contrast,
each of the DISTILLER modules contains in general much less genes than
module `sodA\_c3\_g.75'. Genes in these modules show less extreme
expression levels indicating a weaker coexpression signal compared to
`sodA\_c3\_g.75'.  

When evaluating the overlap between modules found by
these two methods, module 3 and `sodA\_c3\_g.75' share four genes 
(\textit{aldA}, \textit{acnB}, \textit{sdhC}, \textit{sodA}) and 19 condition 
contrasts (`response to oxygen level', `growth').  
These contrasts represent approximately 25\% and 35\%
of all condition contrasts of each module respectively. Module 8 and
`sodA\_c3\_g.75' have three genes (\textit{cyoA}, \textit{sdhC}, \textit{sodA}) 
and eight condition contrasts (`response to oxygen level') in common, which 
represent approximately 6\% and 15\% of all condition contrasts in each module. 
This kind of similarity can be expected, since we applied both methods to
answer same biological question based on the same expression data.

On the other hand, no significant overlap is observed between module 2
and `sodA\_c3\_g.75'. Module 12 shares three genes (\textit{aldA}, 
\textit{cyoA}, \textit{sodA}) with `sodA\_c3\_g.75' though, but the two modules 
have no condition contrasts in common. 
Upon closer inspection of the expression data, genes in
DISTILLER module 2 are coexpressed, but the expression value of \textit{sodA} is
not very high for these condition contrasts, when compared with those of
`sodA\_c3\_g.75'. Hence those condition contrasts will not be easily
picked up by COLOMBOS in Step 2.3 of Section \ref{sec:dist-module-colombos} 
(unless a much lower selection threshold is used). 
Furthermore, genes appearing only in the DISTILLER modules show a different 
expression behavior from that of \textit{sodA} under those condition contrasts 
selected by COLOMBOS. 
As a result, when extending genes of a module based on expression profiles (see 
Step 4 of Section \ref{sec:dist-module-colombos}), those genes will be ranked 
less relevant. Consequently, they will hardly be considered as prime candidates 
for module extension.

In summary, COLOMBOS focuses on using expression values as its main
criterion for building modules, resulting in clear expression changes in
tight coexpression. It can easily extract prominent coexpression
behaviour in the expression data, but might miss modules with less
significant coexpression patterns. The coexpression patterns retrieved
by COLOMBOS can be broadly interpreted as genes being functionally
related as their abundance are altered in similar ways in response to
various stimuli. This functional relationship could imply coregulation
(which might be identified using motif detection algorithms, see also
Fig. \ref{fig:workflow-distiller-colombos}), but it is not a necessary 
prerequisite. 
Instead, it cannot be excluded that several regulatory programs might be 
responsible for the observed coexpression patterns.
On the other hand, DISTILLER, as a global method, tries to recover
distinctive modules that can be directly linked to a shared regulatory
program, i.e. of which the genes are coregulated by the same (set of)
regulator(s). Combining motif information with expression data, it
successfully retrieves less prominent coexpression patterns with
biological significance by utilizing extra information regarding the
presence (or prediction, depending on the nature of the motif input
data) of transcription factor binding sites in its genes promoter
regions. The case study of \textit{sodA} presented in this chapter illustrates 
this complementarity of these two approaches for retrieving biologically 
relevant results. 


Note, in this chapter, we focus on the specific application of the COLOMBOS 
and DISTILLER.  However, the versatile tools as them have many other 
functionalities and application domains.  
%
In COLOMBOS, various options exist to build the module based on the information 
different from a given set of genes.  Furthermore, if desirable, the module 
data can be downloaded for further analysis.  As an example, the data of the 
module `sodA\_c3\_g.75' is exported into file 
`expdata\_COLOMBOS\_module\_information.txt' (available as a sample file, see 
Section \ref{sec:dist-sample}).
%
DISTILLER can use other data sources as input, for instance, the information 
on the binding of regulators as obtained from ChIP-chip experiments 
\cite{Lemmens2009, Lemmens2006}.
Furthermore, it can be used together with data source other than motif data to 
discover other type of networks, as long as connections exist between the data 
source and the gene expression profile. 
For instance, protein-protein interaction data or synthetic lethality data can 
be included. \todo{Still need to review this part.}



%Here we only briefly described the functionalities of COLOMBOS related to 
%constructing modules based only on a known gene set. However, various options 
%exist to build the module based on different types of information. It is also 
%possible to download the data of a module to further analyze 
%it (see Appendix \ref{apd:data-download}). As an example, the data of the 
%module `sodA\_c3\_g.75' obtained in here is exported into file 
%`expdata\_COLOMBOS\_module\_information.txt' (available 
%as a sample file, see Section \ref{sec:dist-sample}). 
%Interested users can refer to Appendix \ref{apd:colombos} for more 
%information. 
%% %
%Here we only discussed how to use DISTILLER to identify transcription 
%networks. However, a versatile tool like DISTILLER has many other application 
%domains. 
%It can use other data sources as input, for instance, the information 
%on the binding of regulators as obtained from ChIP-chip experiments 
%\cite{Lemmens2009, Lemmens2006}.
%Furthermore, it can be used together with data source other than motif data to 
%discover other type of networks, as long as connections exist between the data 
%source and the gene expression profile. 
%For instance, protein-protein interaction data or synthetic lethality data can 
%be included. 
%Binary data is preferred; for other types of data, the preprocessing step can 
%convert them into binary ones.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Keep the following \cleardoublepage at the end of this file, 
% otherwise \includeonly includes empty pages.
\cleardoublepage
 
% vim: tw=70 nocindent expandtab foldmethod=marker foldmarker={{{}{,}{}}}
